plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -193,18 +193,34 @@ public function update( &$order ) {
 		// Update the order.
 		parent::update( $order );
 
-		// Fire a hook depending on the status - this should be considered a creation if it was previously draft status.
 		$new_status = $order->get_status( 'edit' );
 
-		if ( $new_status !== $previous_status && in_array( $previous_status, array( 'new', 'auto-draft', 'draft', 'checkout-draft' ), true ) ) {
+		// We need to remove the wc- prefix from the status for comparison and proper evaluation of new vs updated orders.
+		if ( strpos( $previous_status, 'wc-' ) === 0 ) {
+			$previous_status = substr( $previous_status, 3 );
```

Comment:
nit: `current_status` might be more understandable. We're looking at current status from the DB before the order object is updated.

Reply:
done in https://github.com/woocommerce/woocommerce/pull/49098/commits/146b14297abba200d644cfda31bd59a52ec059e0 👍🏼